modules/k8s/synapse-ha/main.tf

```diff
@@ -0,0 +1,164 @@
+terraform {
+  required_version = ">= 1.6.6"
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = ">= 0.10.1"
+    }
+  }
+}
+
+resource "juju_model" "synapse" {
```

Comment:
WDYT about using the juju_model datasource to import an existing model?

Reply:
I am seeing that the rest of the modules are relying on this configured via a variable, so I'm removing model creation and  will refer to it via the var for consistency. Is that fine?